CRM/Member/ActionMapping.php

```diff
@@ -91,12 +91,15 @@ public function createQuery($schedule, $phase, $defaultParams) {
     }
 
     // FIXME: Numbers should be constants.
+    $contribution_status_id_pending = CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Pending');
     if (in_array(2, $selectedStatuses)) {
       //auto-renew memberships
-      $query->where("e.contribution_recur_id IS NOT NULL");
+      $query->join('contribution_recur', 'inner join civicrm_contribution_recur on e.contribution.id = contribution_recur.id');
+      $query->where("contribution_recur.contribution_status_id = #pending", ['pending' => $contribution_status_id_pending]);
```

Comment:
@stesi561 I'm thinking we might need to handle In Progress as well as pending in case it is ongoing? maybe we should just do an exclusionary list perhaps?

Reply:
yes this is very WIP I'm not sure what these statuses mean tbh! :)